### ISSUE TITLE
Theme is not applied to styled components props in Vitest

### ISSUE THREAD

**Issue by hpinkos** [mantinedev/mantine#4265]
### What package has an issue

@mantine/core

### Describe the bug

I'm using Mantine in a vite/react app.  I found a bug where the `theme` does not appear to be passed correctly to an emotion styled component when running unit tests with Vitest.  In this case, `theme` is `{}`, resulting in the unit test to fail.
For reference, it looks like emotion fixed a similar bug last year: https://github.com/emotion-js/emotion/issues/2955

When I wrap the component in a `MantineProvider` with the theme, the unit test crashes
![image](https://github.com/mantinedev/mantine/assets/3451886/00129500-9c24-4911-9e66-53a2d262dca1)

However, when I wrap the component in an emotion `ThemeProvider` the test passes.

I've created a test case in codesandbox that demonstrates
- The component works as expected when running in the browser with a `MantineProvider`
- A failing unit test when using a `MantineProvider` to set up the theme
- A passing unit test when using a `ThemeProvider` to set up the theme

### What version of @mantine/hooks page do you have in package.json?

6.0.11

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/p/github/hpinkos/mantine-vitest-bug/main

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

No

**Comment by vitoraraujodev** [comment]
hey there

The answer to your question is on this StackOverflow thread right [here](https://stackoverflow.com/a/58302876/13535741) :ok_hand: 

it worked for me ;)

**Comment by oscar-ferrandiz** [comment]
Same issue here 😢 

Any update? 🙏

**Comment by rtivital** [comment]
This issue is closed for one of these reasons:

- It was fixed in 7.0
- It is no longer reproducible in 7.0
- It is no longer applicable in 7.0
- It does not have a reproduction

If you think that this issue was closed by mistake, and it is still an issue in version 7.0, please reopen it.

**Comment by mnai01** [comment]
I encountered the same issue with version 7 despite following the documentation. It seems the documentation is missing a crucial part about specifying the theme in the `emotion.d.ts` file. Here's the fix that worked for me:

```emotion.dts```

```ts
import "@mantine/core";
import "@emotion/react";
import type { EmotionStyles, EmotionSx } from "@mantine/emotion";
import type { MantineTheme } from "@mantine/core";

// Add this
declare module "@emotion/react" {
  export interface Theme extends MantineTheme {}
}

// This line is already added in the docs as seen here https://mantine.dev/styles/emotion/#usage-with-vite
declare module "@mantine/core" {
  export interface BoxProps {
    sx?: EmotionSx;
    styles?: EmotionStyles;
  }
}

```

**Comment by mnai01** [comment]
> This issue is closed for one of these reasons:
> 
> * It was fixed in 7.0
> * It is no longer reproducible in 7.0
> * It is no longer applicable in 7.0
> * It does not have a reproduction
> 
> If you think that this issue was closed by mistake, and it is still an issue in version 7.0, please reopen it.

If the proper fix is to add export interface Theme extends MantineTheme, then yes, it seems to be fixed. However, if this is considered a hacky workaround, then the issue is not yet resolved.
